pro/src/pages/SignIn/SignIn.tsx

```diff
@@ -33,6 +52,50 @@ const SignIn = (): JSX.Element => {
     }
   }, [searchParams])
 
+  const onSubmit = (
+    values: SigninFormValues,
+    submitting: (x: boolean) => void
+  ) => {
+    const { email, password } = values
+    api
+      .signin({ identifier: email, password })
+      .then(user => {
+        dispatch(setCurrentUser(user ? user : null))
+      })
+      .catch(payload => {
+        setCurrentUser(null)
+        onHandleFail(
+          { status: payload.status, errors: payload.body },
+          submitting
+        )
+      })
+  }
+
+  const formik = useFormik({
+    initialValues: SIGNIN_FORM_DEFAULT_VALUES,
+    onSubmit: (values, { setSubmitting }) => onSubmit(values, setSubmitting),
+    validationSchema,
+    validateOnChange: true,
+  })
+
+  const onHandleFail = (
+    payload: SigninApiErrorResponse,
+    submitting: (x: boolean) => void
+  ) => {
+    const { errors, status } = payload
+    if (status === HTTP_STATUS.TOO_MANY_REQUESTS) {
+      notify.error(
+        'Nombre de tentatives de connexion dépassé. Veuillez réessayer dans 1 minute.'
+      )
+    } else if (errors && Object.values(errors).length > 0) {
+      notify.error('Identifiant ou mot de passe incorrect.')
+      formik.setStatus('apiError')
+      formik.setFieldError('email', 'Identifiant ou mot de passe incorrect.')
+      formik.setFieldError('password', 'Identifiant ou mot de passe incorrect.')
+    }
+    submitting(false)
```

Comment:
Pourquoi est-ce qu'on a besoin de mettre submitting à false ? Formik ne le fait pas tout seul ? J'ai l'impression que ça complexifie le code de passer cette fonction de handler en handler

[IMPORTANT: If onSubmit is async, then Formik will automatically set isSubmitting to false on your behalf once it has resolved. This means you do NOT need to call formikBag.setSubmitting(false) manually. However, if your onSubmit function is synchronous, then you need to call setSubmitting(false) on your own](https://formik.org/docs/api/formik#onsubmit-values-values-formikbag-formikbag--void--promiseany)

=> j'ai l'impression que c'est parce qu'on a pas la syntaxe async/await